### DISCUSSION TITLE
getServerSideProps causing 504 Gateway Time-out on AWS Elastic Beanstalk

### DISCUSSION THREAD

**Discussion by abcd-ca** [vercel/next.js#12860] [Help]
I need to host on AWS for a project because I am required to restrict to a region that Vercel unfortunately doesn't have an origin region for :(

I'm using Elastic Beanstalk (EB) with Node.js which uses nginx. 

I've got one route `/profile` which uses `getServerSideProps` to fetch some profile data. My build target is empty (default is "server"). When I do `npm run build` (`next build`) then `npm start` (`next start`), it works great locally. On EB my pages that don't use `getServerSideProps` work great but not the one that does. You can see from the screenshot that it's trying to load the props that `getServerSideProps` returns from a url ending in `profile.json`. I noticed that if in my _local_ terminal I try to find that file (or even `./next/data` ) it doesn't exist, so it seems that the next server provides some dynamic serving of this stuff which makes sense I think.

The data that my page is trying to fetch in `getServerSideProps` is an api within the same app. If I use curl or Postman to fetch the data directly from the api on EB it does retrieve it, instantly. So, I don't believe this is due to some long running process that is bumping up against the default timeout setting in nginx. Rather, I have a feeling that it has something to do with the mechanism that serves the `.next/data` part that isn't working and I'm not sure what to attribute that to. 

Any ideas?

P.S.

I'm just trying to get a basic proof of concept up on AWS to prove that I can use Next.js there. I realize that I probably don't need to use `getServerSideProps` for this profile use case but I'd like to get to the bottom of why it's not working so I can use it if I come up with a need, like for certain SEO-important pages.

![image](https://user-images.githubusercontent.com/1296103/81899411-a297fe80-956f-11ea-9935-3725abc4fb7d.png)

**Comment by gastonrey** [comment]
Ey did you get any clue about it? am having some issues with Kubernetes on AWS and the serverSideProps() but not always, a little bit confused...

**Reply by thiagomorales** [reply]
> Ey did you get any clue about it? am having some issues with Kubernetes on AWS and the serverSideProps() but not always, a little bit confused...

Same here! 
Nextjs 9.5.2 using `getServerSideProps` in Kubernetes on AWS.. every ~ 1k requests I get about 15-20 `504 Gateway timeout`..

My ELB logs capture that:

![Screen Shot 2020-10-08 at 00 52 00](https://user-images.githubusercontent.com/4661217/95412920-7f942f80-0900-11eb-8288-ff5fd734ab31.png)

Sometime, it's hangs in getting `_next/data/build/*.json` _(where in theory getServerSideProps logic is executed)_, sometimes it's fail in main index page.

![image](https://user-images.githubusercontent.com/4661217/95413228-427c6d00-0901-11eb-97c1-3b78f6bcd659.png)

I'm investigating about keep alive timeouts _VS_ ELB timeout, but so far I did not reach any conclusion.

**Comment by abcd-ca** [comment]
I abandoned Next and went with an Express and React stack on AWS instead, sorry.

**Reply by danroestorf** [reply]
Good choice, Next is nothing but a headache when running anywhere but Vercel.

**Comment by VivienGiraud** [comment]
Same problem here !

**Comment by nicacioliveira** [comment]
Hello. 

There is some possibility of exposing nodejs on a customized server so that it is possible to configure keepalive? (was quoted [here](https://github.com/vercel/next.js/discussions/16544))

I need to set the js node timeout to the default value of aws (60s).

My application has a relevant number of requests and I am having a lot of problems with this and I don't know what else to do...

@thiagomorales Did you get any way out?

**Reply by danroestorf** [reply]
https://github.com/vercel/next.js/discussions/19618#discussioncomment-1344475
